core/src/main/kotlin/xtdb/util/Hasher.kt

```diff
@@ -0,0 +1,44 @@
+package xtdb.util
+
+import org.apache.commons.codec.digest.MurmurHash3
+import org.apache.commons.codec.digest.XXHash32
+import java.nio.ByteBuffer
+
+
+interface Hasher {
+
+    fun hash(value: ByteArray): Int
+    fun hash(value: Double): Int
+    fun hash(value: Long): Int
+
+    abstract class DefaultHasher : Hasher {
```

Comment:
yep, fair - if we need this we might as well make Hasher the abstract class instead, save the level of indirection